save-core/src/commonMain/kotlin/org/cqfn/save/core/config/SaveConfig.kt

```diff
@@ -22,23 +23,161 @@ import okio.Path
  * @property configInheritance whether configuration files should inherit configurations from the previous level of directories
  * @property ignoreSaveComments if true, ignore technical comments, that SAVE uses to describe warnings, when running tests
  * @property reportDir path to directory where to store output (when `resultOutput` is set to `FILE`)
+ * @property propertiesFile path to the file with extra properties (that can be also passed as command-line arguments)
  */
-@OptIn(ExperimentalFileSystem::class)
-data class SaveConfig(
-    val configPath: Path,
-    val parallelMode: Boolean,
-    val threads: Int,
-    val propertiesFile: Path,
-    val debug: Boolean,
-    val quiet: Boolean,
-    val reportType: ReportType,
-    val baselinePath: Path?,
-    val excludeSuites: List<String>,
-    val includeSuites: List<String>,
-    val language: LanguageType,
-    val testRootPath: Path,
-    val resultOutput: ResultOutputType,
-    val configInheritance: Boolean,
-    val ignoreSaveComments: Boolean,
-    val reportDir: Path,
-)
+@Serializable
+class SaveConfig(
+    var configPath: String? = "save.toml",
+    var parallelMode: Boolean? = false,
+    var threads: Int? = 1,
+    var debug: Boolean? = false,
+    var quiet: Boolean? = false,
+    var reportType: ReportType? = ReportType.JSON,
+    var baseline: String? = null,
+    var excludeSuites: String? = null,
+    var includeSuites: String? = null,
+    var language: LanguageType? = LanguageType.UNDEFINED,
+    var testRootPath: String? = null,
+    var configInheritance: Boolean? = true,
+    var ignoreSaveComments: Boolean? = false,
+    var reportDir: String? = "save-reports",
+    var resultOutput: ResultOutputType? = ResultOutputType.STDOUT,
+    var propertiesFile: String? = null,
+) {
+    constructor(args: Array<String>) : this() {
+        val parser = ArgParser("save")
+
+        configPath = parser.option(
+            ArgType.String,
+            fullName = "config",
+            shortName = "c",
+            description = "Path to a configuration of a test suite",
+        ).value
+
+        parallelMode = parser.option(
+            ArgType.Boolean,
+            fullName = "parallel-mode",
+            shortName = "parallel",
+            description = "Whether to enable parallel mode",
+        ).value
+
+        threads = parser.option(
+            ArgType.Int,
+            fullName = "threads",
```

Comment:
Previously, when we initialized in form `val threads by parser.option(...)`, we needed fullName only in specific cases.\
Usually, only if the name of variable was in cameCase.
Do we need now to duplicate the name, since we initialize them in other manner?

Reply:
yes, we need this name now